index.js

```diff
@@ -14,226 +11,207 @@ const getDirName = require('path').dirname;
-            }
-        });
-
-        fs.access(configuration.baseFolder, fs.constants.F_OK | fs.constants.W_OK, (err) => {
-            if (err) {
-                console.error(
-                    `${configuration.baseFolder} ${err.code === 'ENOENT' ? 'does not exist' : 'is read-only'}`);
-            }
-        });
-
-        fs.copyFileSync(configuration.screenshotFolder + screenShotImage, configuration.baseFolder + screenShotImage);
+    options.boundingBox = await this._getBoundingBox(selector);
+    const misMatch = await this._fetchMisMatchPercentage(baseImage, options);
+    this.debug("MisMatch Percentage Calculated is " + misMatch);
+    assert(misMatch <= options.tolerance, "MissMatch Percentage " + misMatch);
+  }
+
+  /**
+   * Function to prepare Base Images from Screenshots
+   *
+   * @param screenShotImage  Name of the screenshot Image (Screenshot Image Path is taken from Configuration)
+   */
+  async _prepareBaseImage(screenShotImage) {
+    const configuration = this.config;
+
+    await this._createDir(configuration.baseFolder + screenShotImage);
+
+    fs.access(configuration.screenshotFolder + screenShotImage, fs.constants.F_OK | fs.constants.W_OK, (err) => {
+      if (err) {
+        throw new Error(
+          `${configuration.screenshotFolder + screenShotImage} ${err.code === 'ENOENT' ? 'does not exist' : 'is read-only'}`);
+      }
+    });
+
+    fs.access(configuration.baseFolder, fs.constants.F_OK | fs.constants.W_OK, (err) => {
+      if (err) {
+        throw new Error(
+          `${configuration.baseFolder} ${err.code === 'ENOENT' ? 'does not exist' : 'is read-only'}`);
+      }
+    });
+
+    fs.copyFileSync(configuration.screenshotFolder + screenShotImage, configuration.baseFolder + screenShotImage);
+  }
+
+  /**
+   * Function to create Directory
+   * @param directory
+   * @returns {Promise<void>}
+   * @private
+   */
+  _createDir(directory) {
+    mkdirp.sync(getDirName(directory));
+  }
+
+  /**
+   * Function to fetch Bounding box for an element, fetched using selector
+   *
+   * @param selector CSS|XPath|ID selector
+   * @returns {Promise<{boundingBox: {left: *, top: *, right: *, bottom: *}}>}
+   */
+  async _getBoundingBox(selector) {
+    const helper = this._getHelper();
+    await helper.waitForVisible(selector);
+    const els = await helper._locate(selector);
+    if (!els.length) throw new Error(`Element ${selector} couldn't be located`);
+    const el = els[0];
+
+    let location, size;
+
+    if (this.helpers['Puppeteer']) {
+      const box = await el.boundingBox();      
+      size = location = box;
     }
 
-    /**
-     * Function to create Directory
-     * @param directory
-     * @returns {Promise<void>}
-     * @private
-     */
-    async _createDir (directory) {
-        mkdirp.sync(getDirName(directory));
+    if (this.helpers['WebDriver'] || this.helpers['Appium']) {
+      location = await ele.getLocation();
```

Comment:
@DavertMik could you please change this `ele` to `el` because, ele is no longer defined and returns an error, it should be `el`